orttraining/orttraining/core/graph/graph_augmenter.cc

```diff
@@ -40,45 +75,34 @@ Status GraphAugmenter::AugmentGraph(Graph& graph, const GraphDefs& graph_element
                   node_def.domain);
   }
 
-  // Add new outputs to the graph.
-  std::vector<const NodeArg*> new_output_args = graph.GetOutputs();  // Make a copy of existing output args.
-  for (const auto& output_name : graph_element_defs.GraphOutputs()) {
-    const auto* output_arg = graph.GetNodeArg(output_name);
+  // Add new inputs to the graph.
+  if (!graph_element_defs.GraphInputs().empty()) {
+    std::vector<const NodeArg*> new_inputs;
+    ORT_RETURN_IF_ERROR(AddToExistingNodeArgs(
+        "add graph inputs", graph, graph_element_defs.GraphInputs(), graph.GetInputsIncludingInitializers(),
+        false, new_inputs));
+    graph.SetInputs(new_inputs);
+  }
 
-    ORT_RETURN_IF(output_arg == nullptr, "Failed to set graph output ", output_name);
-    if (std::find(new_output_args.begin(), new_output_args.end(), output_arg) == new_output_args.end()) {
-      new_output_args.emplace_back(output_arg);
-    }
+  // Add new outputs to the graph.
+  if (!graph_element_defs.GraphOutputs().empty()) {
+    std::vector<const NodeArg*> new_outputs;
+    ORT_RETURN_IF_ERROR(AddToExistingNodeArgs(
+        "add graph outputs", graph, graph_element_defs.GraphOutputs(), graph.GetOutputs(), false, new_outputs));
+    graph.SetOutputs(new_outputs);
   }
-  graph.SetOutputs(new_output_args);  // By setting this, Graph::SetGraphInputsOutputs could infer the output as expected.
 
   graph.SetGraphResolveNeeded();
-  graph.SetGraphProtoSyncNeeded();
   return graph.Resolve();
 }
 
 Status GraphAugmenter::OverrideGraphOutputs(Graph& graph, const std::vector<std::string>& graph_outputs) {
-  {
-    std::unordered_set<std::string> unique_graph_outputs(graph_outputs.begin(), graph_outputs.end());
-    if (unique_graph_outputs.size() != graph_outputs.size()) {
-      std::string error_message{"The specified graph outputs are not unique:"};
-      for (const auto& graph_output : graph_outputs) {
-        error_message += "\n  " + graph_output;
-      }
-      return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, error_message);
-    }
-  }
```

Comment:
Is this check no longer needed? Having duplicate names in the output can cause issues